### PR TITLE
Change update location summaries to query by state rather than pulling counties in all at once

### PR DESCRIPTION
**Pull request by ghop02**
We finally hit the limit of string size for parsing a json file for the version of node we're on.  This splits up the querying by states - we should be good for a while (and maybe can even upgrade our version of node now!)

### CODE REVIEW

File: scripts/county_history_data/generate.ts [hunk 1]
```diff
@@ -8,11 +8,11 @@
 import fs from 'fs-extra';
 import path from 'path';
 import _ from 'lodash';
-import { fetchAllCountyProjections } from '../../src/common/utils/model';
+import { fetchAllCountyProjectionsByState } from '../../src/common/utils/model';
 import { Projection } from '../../src/common/models/Projection';
 
 async function main() {
-  const allProjections = await fetchAllCountyProjections();
+  const allProjections = await fetchAllCountyProjectionsByState();
```
Comment (mikelehen) [inline scripts/county_history_data/generate.ts:15]: I think you can undo this change?

File: src/common/utils/model.ts [hunk 2]
```diff
@@ -100,7 +100,14 @@ export function fetchAllStateProjections(snapshotUrl: string | null = null) {
 
 /** Returns an array of `Projections` instances for all counties. */
 const cachedCountiesProjections: { [key: string]: Promise<Projections[]> } = {};
-export function fetchAllCountyProjections(snapshotUrl: string | null = null) {
+export function fetchAllCountyProjections(
+  snapshotUrl: string | null = null,
+  queryByState: boolean = true,
```
Comment (mikelehen) [inline src/common/utils/model.ts:105]: I don't think we really need this flag.  We can just merge these methods and always query by state.

File: src/common/utils/model.ts [hunk 3]
```diff
@@ -123,6 +130,48 @@ export function fetchAllCountyProjections(snapshotUrl: string | null = null) {
...
+  state: State,
+  snapshotUrl: string | null = null,
+) {
+  snapshotUrl = snapshotUrl || getSnapshotUrlOverride();
+  async function fetch() {
+    const all = await new Api(
+      snapshotUrl,
+    ).fetchCountySummariesWithTimeseriesForState(state);
+    return all
+      .filter(summaryWithTimeseries => {
```
Comment (mikelehen) [inline src/common/utils/model.ts:158]: I think we can drop this filter because this function should never be called for states not in our regions instance (this filtering originally showed up when e.g. we started adding territories to the backend but the frontend didn't know about them).

File: src/api/index.ts [hunk 4]
```diff
@@ -65,6 +65,19 @@ export class Api {
     return all;
   }
 
+  /**
+   * Fetches the summary+timeseries for every region in the specified state
```
Comment (mikelehen) [inline src/api/index.ts:69]: region=>county ?

**Review by mikelehen — APPROVED** [review]
Thanks for knocking this out.  Would like a few small cleanups, but overall looks good.

File: src/common/utils/model.ts (hunk 3 above)
Comment (ghop02) [reply src/common/utils/model.ts:158]: This is actually filtering counties not states - I'd lean towards keeping it just to make sure we don't include counties that  aren't matching 1:1 (or we can continue with the strict call).  but can remove if you feel strongly

Comment (mikelehen) [reply src/common/utils/model.ts:158]: Oh, I see what you mean.  Yeah, I guess I don't feel super strongly but I also can't immediately think of a case where the backend would have extra counties for a state and we would want to silently ignore that, so I probably lean towards removing this code, which it looks like you did in the follow-up PR.  Thanks!
